src/test/java/rx/util/async/AsyncTest.java

```diff
@@ -819,6 +838,500 @@ public String answer(InvocationOnMock invocation) throws Throwable {
 
         verify(func, times(1)).call();
     }
+    
+    @Test
+    public void testStartCallable() throws Exception {
+        TestScheduler scheduler = new TestScheduler();
+        Callable<String> func = Mockito.mock(Callable.class);
+        Mockito.when(func.call()).thenReturn("One");
+        
+        Observable<String> observable = Async.startCallable(func, scheduler);
+        
+        scheduler.advanceTimeBy(100, TimeUnit.MILLISECONDS);
+        
+        Observer<String> observer1 = Mockito.mock(Observer.class);
+        Observer<String> observer2 = Mockito.mock(Observer.class);
+        observable.subscribe(observer1);
+        observable.subscribe(observer2);
+        
+        InOrder inOrder = Mockito.inOrder(func, observer1, observer2);
+        
+        inOrder.verify(func, times(1)).call(); // verify the function is only called once, not each time someone subscribes.
+        inOrder.verify(observer1, times(1)).onNext("One");
+        inOrder.verify(observer1, times(1)).onCompleted();
+        inOrder.verify(observer2, times(1)).onNext("One");
+        inOrder.verify(observer2, times(1)).onCompleted();
+        inOrder.verifyNoMoreInteractions();
+    }
+    
+    @Test
+    public void testStartCallableThrows() throws Exception {
+        TestScheduler scheduler = new TestScheduler();
+        Exception exception = new Exception();
+        Callable<String> func = Mockito.mock(Callable.class);
+        Mockito.when(func.call()).thenThrow(exception);
+        
+        Observable<String> observable = Async.startCallable(func, scheduler);
+        
+        scheduler.advanceTimeBy(100, TimeUnit.MILLISECONDS);
+        
+        Observer<String> observer1 = Mockito.mock(Observer.class);
+        Observer<String> observer2 = Mockito.mock(Observer.class);
+        observable.subscribe(observer1);
+        observable.subscribe(observer2);
+        
+        InOrder inOrder = Mockito.inOrder(func, observer1, observer2);
+        
+        inOrder.verify(func, times(1)).call(); // verify the function is only called once, not each time someone subscribes.
+        inOrder.verify(observer1, times(1)).onError(exception);
+        inOrder.verify(observer2, times(1)).onError(exception);
+        inOrder.verifyNoMoreInteractions();
+    }
+    
+    @Test
+    public void testCallable() throws Exception {
+        Callable<Integer> func = spy(new Callable<Integer>() {
+            @Override
+            public Integer call() {
+                return 0;
+            }
+        });
+        Observable<Integer> observable = 
+        	Async.toAsync(func, Schedulers.immediate())
+                     .call();
+        
+        // Verifies that the returned Observable is "hot"
+        verify(func, times(1)).call();
+        
+        observable.subscribe(new TestObserver<Object>(observer));
```

Comment:
Why do you wrap `observer` into a `TestObserver`? This goes for all the other places below.